packages/web-app-files/src/components/SideBar/Shares/Collaborators/RoleDropdown.vue

```diff
@@ -237,9 +242,11 @@ export default defineComponent({
       } else if (this.resourceIsSpace) {
         this.selectedRole = SpacePeopleShareRoles.list()[0]
       } else {
+        const canDeny = this.resource.canDeny() && this.hasRoleDenyAccess
         this.selectedRole = PeopleShareRoles.list(
           this.resource.isFolder,
-          this.resource.canDeny()
+          this.hasRoleCustomPermissions,
```

Comment:
This was actually a bug, because `canDeny` was passed as `hasCustom` param for the `list` method.